core/src/main/java/nu/yona/server/goals/service/GoalService.java

```diff
@@ -225,8 +229,7 @@ private void deleteGoalRelatedMessages(UserAnonymized userAnonymizedEntity, Goal
 	private void deleteGoalConflictMessagesForGoal(UserAnonymized userAnonymizedEntity, Goal goalEntity)
 	{
 		userAnonymizedEntity.getAnonymousDestination().removeGoalConflictMessages(goalEntity);
-		userAnonymizedEntity.getBuddiesAnonymized()
-				.forEach(ba -> ba.getUserAnonymized().getAnonymousDestination().removeGoalConflictMessages(goalEntity));
+		buddyService.getBuddyDestinations(userAnonymizedEntity).forEach(d -> d.removeGoalConflictMessages(goalEntity));
```

Comment:
When there are messages sent to buddies that are now disconnected, are they removed ever? Or are they just kept?